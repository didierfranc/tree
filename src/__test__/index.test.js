const data = require('./data.json')
const { toTree, toFlatTree } = require('../')

it('generate tree from array', () => {
  const tree = toTree(data)
  expect(tree).toMatchSnapshot()
})

it('add depth to array element', () => {
  const tree = toFlatTree(data)
  console.log(tree)
  expect(tree).toMatchSnapshot()
})
