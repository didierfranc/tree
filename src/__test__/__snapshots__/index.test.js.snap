// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add depth to array element 1`] = `
Array [
  Object {
    "depth": 0,
    "id": "1",
    "parent": null,
  },
  Object {
    "depth": 0,
    "id": "2",
    "parent": null,
  },
  Object {
    "depth": 1,
    "id": "4",
    "parent": "2",
  },
  Object {
    "depth": 2,
    "id": "5",
    "parent": "4",
  },
  Object {
    "depth": 2,
    "id": "6",
    "parent": "4",
  },
  Object {
    "depth": 3,
    "id": "7",
    "parent": "6",
  },
  Object {
    "depth": 0,
    "id": "3",
    "parent": null,
  },
]
`;

exports[`generate tree from array 1`] = `
Array [
  Object {
    "children": Array [],
    "id": "1",
    "parent": null,
  },
  Object {
    "children": Array [
      Object {
        "children": Object {
          "children": Array [
            Object {
              "children": Object {
                "children": Array [],
                "id": "5",
                "parent": "4",
              },
              "depth": 2,
              "id": "5",
              "parent": "4",
            },
            Object {
              "children": Object {
                "children": Array [
                  Object {
                    "children": Object {
                      "children": Array [],
                      "id": "7",
                      "parent": "6",
                    },
                    "depth": 3,
                    "id": "7",
                    "parent": "6",
                  },
                ],
                "id": "6",
                "parent": "4",
              },
              "depth": 2,
              "id": "6",
              "parent": "4",
            },
          ],
          "id": "4",
          "parent": "2",
        },
        "depth": 1,
        "id": "4",
        "parent": "2",
      },
    ],
    "id": "2",
    "parent": null,
  },
  Object {
    "children": Array [],
    "id": "3",
    "parent": null,
  },
]
`;
